src/components/App/App.js

```diff
@@ -105,30 +117,77 @@ function App() {
   }
 
   // ********** Submission Handlers **********
-  function handleLoginSubmit(user) {
+  function handleLoginSubmit(email, password) {
     setIsLoading(true);
-    if (user.email === "user@host.com" && user.password === "password") {
-      const user = database.users[0];
-      setCurrentUser(user);
-      setAlternateAvatar(getUserFirstLetter(user.name));
-      setIsLoggedIn(true);
-      closeModal();
-    } else {
-      setErrorDisplay({ value: true, message: "Invalid email or password." });
-    }
-    setIsLoading(false);
+
+    const user = { email, password };
+
+    signin(user)
+      .then((res) => {
+        localStorage.setItem("token", res.token);
+
+        checkToken(res.token).then((res) => {
+          setCurrentUser(JSON.parse(JSON.stringify(res.data)));
+          setIsLoggedIn(true);
+          setAlternateAvatar(getUserFirstLetter(res.data.name));
+          setActiveMenuSelection(userDropdown[0]);
+          history.push("/");
+        });
+
+        closeModal();
+      })
+      .catch((err) => {
+        handleModalErrorDisplay(true, errorMessageHandler(err));
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
   }
 
-  function handleSignUpSubmit() {
-    history.push("/building");
-    closeModal();
+  function handleSignUpSubmit(values) {
+    setIsLoading(true);
+
+    const { email, password } = values;
+
+    signup(values)
+      .then((res) => {
+        handleLoginSubmit(email, password);
+        closeModal();
+      })
+      .catch((err) => {
+        handleModalErrorDisplay(true, errorMessageHandler(err));
+      })
+      .finally(() => {
```

Comment:
**Might be better**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js
  function handleSubmit(request) {
    // start loading
    setIsLoading(true);
    request()
      // we need to close only in `then`
      .then(closeModal)
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
      .catch(console.error)
      // and in finally we need to stop loading
      .finally(() => setIsLoading(false));
  }

```